ResearchKit/ActiveTasks/ORKToneAudiometryContentView.m

```diff
@@ -0,0 +1,184 @@
+ FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#import "ORKToneAudiometryContentView.h"
+#import "ORKSkin.h"
+#import "ORKUnitLabel.h"
+#import "ORKHelpers.h"
+
+@interface ORKToneAudiometryContentView ()
+
+@property (nonatomic, strong) ORKUnitLabel *captionLabel;
+@property (nonatomic, strong) UIProgressView *progressView;
+
+@end
+
+
+@implementation ORKToneAudiometryContentView {
+    NSArray *_constraints;
+    ORKScreenType _screenType;
+}
+
+- (instancetype)init {
+    self = [super init];
+    if (self) {
+
+        _screenType = ORKScreenTypeiPhone4;
+        _captionLabel = [ORKUnitLabel new];
+        _captionLabel.textAlignment = NSTextAlignmentCenter;
+        _captionLabel.translatesAutoresizingMaskIntoConstraints = NO;
+
+        _progressView = [UIProgressView new];
+        _progressView.translatesAutoresizingMaskIntoConstraints = NO;
+        _progressView.progressTintColor = [self tintColor];
+        [_progressView setAlpha:0];
+
+        _tapButton = [[ORKRoundTappingButton alloc] init];
+        _tapButton.translatesAutoresizingMaskIntoConstraints = NO;
+        [_tapButton setTitle:ORKLocalizedString(@"TAP_BUTTON_TITLE", nil) forState:UIControlStateNormal];
+
+        [self addSubview:_captionLabel];
+        [self addSubview:_progressView];
+        [self addSubview:_tapButton];
+
+        self.translatesAutoresizingMaskIntoConstraints = NO;
+
+        _captionLabel.text = nil;
+
+        [self setNeedsUpdateConstraints];
+    }
+
+    return self;
+}
+
+- (void)tintColorDidChange {
+    [super tintColorDidChange];
+    self.progressView.progressTintColor = [self tintColor];
+}
+
+- (void)setProgress:(CGFloat)progress
+            caption:(NSString *)caption
+           animated:(BOOL)animated {
+    self.captionLabel.text = caption;
+
+    [self.progressView setProgress:progress animated:animated];
+    [UIView animateWithDuration:animated ? 0.2 : 0 animations:^{
+        [self.progressView setAlpha:(progress == 0) ? 0 : 1];
+    }];
+}
+
+- (void)finishStep:(ORKActiveStepViewController *)viewController {
+    [super finishStep:viewController];
+    self.tapButton.enabled = NO;
+}
+
+- (void)willMoveToWindow:(UIWindow *)newWindow {
+    [super willMoveToWindow:newWindow];
+    _screenType = ORKGetScreenTypeForWindow(newWindow);
+    [self setNeedsUpdateConstraints];
+}
+
+- (void)updateConstraints {
+    if ([_constraints count]) {
+        [NSLayoutConstraint deactivateConstraints:_constraints];
+        _constraints = nil;
+    }
+
+    ORKScreenType screenType = _screenType;
```

Comment:
This constraint code does not actually need to be dynamic. Move to a -setupConstraints method, called in the initializer, that creates and activates the constraints. We're gradually changing this elsewhere too, to avoid unnecessary layout re-calculation, but let's fix it here now.